styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java

```diff
@@ -279,6 +297,18 @@ public WorkflowResource(Storage storage, WorkflowValidator workflowValidator,
     return Response.forPayload(data);
   }
 
+  /**
+   * Calculate an offset suitable for getting the {@code limit} latest instances of a workflow.
+   */
+  private String calculateTailOffset(int limit, Schedule schedule) {
+    // TODO: Optimize away for loop using Schedule.WellKnown
+    Instant offset = Instant.now();
+    for (int i = 0; i < limit; i++) {
+      offset = TimeUtil.previousInstant(offset, schedule);
```

Comment:
I remember this is pretty heavy computation in case of cron because of overhead in `previousInstant`. So we might want to put down the loop.

Reply:
Not sure what put down the loop means but at least now it's optimized for "well known" time units.

Reply:
Something like we did for `TimeUtil::instantsInRange()`.